### ISSUE TITLE
Can't install on Firefox

### ISSUE THREAD

**Issue by francislavoie** [FrankerFaceZ/FrankerFaceZ#1495]
**Describe the bug**
I try to install the extension, either by clicking the Download button on the website, or opening the `.xpi`, but in both cases I get an error saying the extension is corrupted.

**To Reproduce**
1. Go to https://www.frankerfacez.com/
2. Click Download (i.e. https://cdn.frankerfacez.com/script/frankerfacez-4.0-an+fx.xpi)
3. See above error

**Expected behavior**
The installation should work.

**Screenshots**

![image](https://github.com/FrankerFaceZ/FrankerFaceZ/assets/2111701/d569b616-02b7-46db-8d9a-9efe915ba4b1)

**Environment (please complete the following information):**
 - OS: Linux (PopOS 22.04)
 - Browser: Firefox Developer Edition 127.0b1
 - Version: N/A, latest `.xpi`
 - Other Extensions: 7TV, uBlock Origin, many others but I doubt that's relevant

**FFZ Logs (via FFZ Control Center > Home > Feedback >> Log)**:

N/A, can't install.

**Additional context**

I tried the steps "You might have an issue with corrupt extension files" the only relevant file I had was `extensions.json`, I deleted and rebooted the browser and there was no change in behaviour. All my other extensions work fine, I can install things from https://addons.mozilla.org/en-CA/firefox/extensions/ just fine.

The extension filename looks suspicious, what does `an+fx` mean?

**Comment by francislavoie** [comment]
Alright, figured it out.

So apparently Firefox made a change to require that all extensions are signed, even ones provided externally. There's the `about:config` flag `xpinstall.signatures.required` which can be toggled to `false`, then installing the extension from the file (I assume also from the website) will work.

So, it seems like the extension needs to be signed from now on. Hopefully you can figure this out for future users, I'm personally satisfied with this workaround for now, but I don't think it's ideal for everyone. Maybe if you can't get signing figured out, you could add the above as a note on the website?

**Comment by electricsteve** [comment]
I got the same issue, please fix this.

**Comment by electricsteve** [comment]
Also the fix you mentioned doesn't work for me.

**Comment by 7e3J4y** [comment]
Also having this problem on a fresh install of Windows 11 with Firefox. Changing both settings related to `xpinstall` to true do not work for getting the extension installed, though I can confirm loading it through about:debugging works as expected. Is there really no other way whatever changed within Firefox can be bypassed so this can be installed permanently?

**Comment by Lordmau5** [comment]
Yes, as mentioned already, if you want to install an unsigned extension in Firefox while keeping it persistent (and not just for one session) you have to use the [Firefox Developer Edition](https://www.mozilla.org/en-US/firefox/developer/) and disabling the signatures required option in the `about:config`.

The better alternative though would be to install a userscript loader such as [Tampermonkey](https://addons.mozilla.org/en-US/firefox/addon/tampermonkey/) and then installing the [FrankerFaceZ Userscript](https://cdn.frankerfacez.com/script/ffz_injector.user.js) from the website (**Other Browsers** below the download button, then **User Script**)

**Comment by robines-wins** [comment]
Seems that the issue is the following error in the browser console : 

`1718183389540	addons.xpi	WARN	Download of https://cdn.frankerfacez.com/static/frankerfacez-4.0-an+fx.xpi failed: install rejected due to the package not including a strong cryptographic signature
`

**Comment by wulf21** [comment]
> The better alternative though would be to install a userscript loader such as [Tampermonkey](https://addons.mozilla.org/en-US/firefox/addon/tampermonkey/) and then installing the [FrankerFaceZ Userscript](https://cdn.frankerfacez.com/script/ffz_injector.user.js) from the website (**Other Browsers** below the download button, then **User Script**)

Thx for the hint, this way of installing worked for me.

**Comment by Lordmau5** [comment]
This was the official announcement from SirStendec on the FFZ Discord

![image](https://github.com/FrankerFaceZ/FrankerFaceZ/assets/1345036/ff706070-7172-48f0-adac-832e3066b6b5)

**Comment by Smoltallk** [comment]
This thread needs to be pinned or better labeled for other firefox user's visibility, I almost missed this thread.

**Comment by ExperimentalCyborg** [comment]
Any updates on this? 

There's no way in hell I'm bypassing browser safety features with userscript nonsense, I'd much rather have a reduced functionality version of the plugin... or not use it at all. Which is my current situation, but having to rely on chatterino isn't the greatest user experience either because of the multi window situation and it not handling combined chats properly.
